addons/account/models/partner.py

```diff
@@ -787,18 +789,19 @@ def can_edit_vat(self):
         )
 
     def write(self, vals):
-        res = super().write(vals)
-        moves_sudo = self.sudo().env['account.move'].search([('partner_id', 'in', self.ids)])
-        if moves_sudo and 'parent_id' in vals:
-            if not self.env.user.has_group('account.group_account_user'):
-                raise UserError(_("You do not have permission to mark this partner as the main commercial partner."))
+        partner2moves = self.sudo().env['account.move'].search([('partner_id', 'in', self.ids)]).grouped('partner_id')
+        if partner2moves and 'parent_id' in vals:
             parent_vat = self.env['res.partner'].browse(vals['parent_id']).vat
             if vals['parent_id'] and {parent_vat} != set(self.mapped('vat')):
                 raise UserError(_("You cannot set a partner as an invoicing address of another if they have a different %(vat_label)s.", vat_label=self.vat_label))
 
-            self._compute_commercial_partner()
-            for partner in self:
-                moves_sudo.filtered(lambda m: m.partner_id == partner)['commercial_partner_id'] = partner.commercial_partner_id.id
+        res = super().write(vals)
+
+        if 'parent_id' in vals:
+            for partner, moves in partner2moves.items():
+                partner._compute_commercial_partner()
+                moves.line_ids.with_context(bypass_lock_check=BYPASS_LOCK_CHECK).partner_id = partner.commercial_partner_id
+                moves.with_context(bypass_lock_check=BYPASS_LOCK_CHECK).commercial_partner_id = partner.commercial_partner_id
```

Comment:
```suggestion
                moves.with_context(bypass_lock_check=BYPASS_LOCK_CHECK).commercial_partner_id = partner.commercial_partner_id
```
lines_ids.partner_id are computed on move_id.commercial_partner_id, so why do we need to write on both? :thinking: